wemake_python_styleguide/logic/nodes.py

```diff
@@ -49,3 +49,29 @@ def get_parent(node: ast.AST) -> Optional[ast.AST]:
 def get_context(node: ast.AST) -> Optional[ast.AST]:
     """Returns the context or ``None`` if node has no context."""
     return getattr(node, 'wps_context', None)
+
+
+SUBNODE_TYPE = TypeVar('SUBNODE_TYPE')
+
+
+def get_subnodes_by_type(
+    node: ast.AST,
+    subnodes_type: Type[SUBNODE_TYPE],
```

Comment:
Should be: `Union[Type[SUBNODE_TYPE], AnyNodes]`
See: https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/types.py#L61

Reply:
In case I specify it like this then Iterator should have got `Union[Type[SUBNODE_TYPE], Type[ast.AST]]` type too which leads to inability to use this generic type on side of invocation of the method. That's why I have decided to restrict it to one type because in my case I wanted to get subnodes with type `ast.Raise` and supply them to function which receives `ast.Raise`. After this change mypy claims that `wemake_python_styleguide.logic.nodes.get_exception_name` expects `ast.Raise` but I supply `Union[Type[SUBNODE_TYPE], Type[ast.AST]]`. Am I doing something wrong?

Declaration:
```python
def get_subnodes_by_type(
    node: ast.AST,
    subnodes_type: Union[Type[SUBNODE_TYPE], AnyNodes],
) -> Iterator[Union[Type[SUBNODE_TYPE], Type[ast.AST]]]:
    """Returns the list of subnodes of given node with given subnode type."""
```

Reply:
Got it 👍